src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -603,27 +604,41 @@ export default class MongooseAdapter {
     return words.join(" ");
   }
 
-  static getIdQueryType(idOrIds) {
-    const mode = typeof idOrIds === "string" ? "findOne" : "find";
-    let idQuery;
+  static getIdQueryType(idOrIds, model) {
+    return Q.Promise((resolve, reject) => {
+      const mode = typeof idOrIds === "string" ? "findOne" : "find";
+      let ids = Array.isArray(idOrIds) ? idOrIds : [ idOrIds ];
+      let idQuery;
+      let tests = [];
 
-    if (typeof idOrIds === "string") {
-      if (!this.idIsValid(idOrIds)) {
-        throw new APIError(404, undefined, "No matching resource found.", "Invalid ID.");
-      }
+      ids = ids.filter(id => Boolean(id));
 
-      idQuery = {_id: idOrIds};
-    }
+      ids.forEach(id => {
+        const testDoc = new model({ _id: id })
+        tests.push(testDoc.validate());
+      });
 
-    else if (Array.isArray(idOrIds)) {
-      if (!idOrIds.every(this.idIsValid)) {
-        throw new APIError(400, undefined, "Invalid ID.");
-      }
+      Q.allSettled(tests).then(results => {
```

Comment:
We shouldn't have to wait for all the validations to complete before determining whether `invalid` is true. Maybe have an outer promise representing the `invalid` flag, and have the first id test that resolves to false reject that outer test.